shell/platform/embedder/embedder.h

```diff
@@ -1632,6 +1653,10 @@ typedef struct {
     // The description of the Vulkan backing store.
     FlutterVulkanBackingStore vulkan;
   };
+
+  /// Extra information set by the engine for the backing store that the
+  /// embedder may use during presentation.
+  FlutterBackingStorePresentInfo* present_info;
```

Comment:
Adding this at the end isn't safe since the offset of this field isn't guaranteed given that both (a) things might get added to this union that make it consume more space and (b) things might get added to structs/unions in the structs in this union (transitively down) that might affect the size of this union.

Reply:
The idea is that adding things to union would produce another version. The logic in embedder_backing_store would ensure that we're not exposing fields outside of current version.

This would get tricky if the backing stores themselves or inner fields change. I assume that was not likely, but I can see in the PR that I was wrong.

We have backed ourself into a corner with the union not having pointer members :-/

I can put the present info back to the `FlutterLayer`. I think ti would be less disruptive than introducing `FlutterBackingStore2`.